client/packages/system/src/Stock/Components/StockLineForm.tsx

```diff
@@ -296,37 +314,57 @@ export const StockLineForm = ({
               />
             }
           />
+          {showItemVariantsInput && (
+            <StyledInputRow
+              label={t('label.item-variant')}
+              Input={
+                <ItemVariantInput
+                  itemId={draft.itemId}
+                  selectedId={draft?.itemVariant?.id}
+                  width={160}
+                  disabled={!packEditable}
+                  onChange={variant => {
+                    const packaging = variant?.packagingVariants.find(
+                      p => p.packSize === draft.packSize
+                    );
+                    const volumePerPack =
+                      ((packaging?.volumePerUnit ?? 0) / 1000) *
+                      (draft?.packSize ?? 1);
+
+                    onUpdate({
+                      itemVariant: variant,
+                      volumePerPack,
+                    });
+                  }}
+                />
+              }
+            />
+          )}
           <StyledInputRow
-            label={t('label.barcode')}
+            label={t('label.volume-per-pack')}
             Input={
-              <Box style={{ width: 162 }}>
-                <BufferedTextInput
-                  value={draft.barcode ?? ''}
-                  onChange={e => onUpdate({ barcode: e.target.value })}
-                />
-                {isEnabled && (
-                  <Tooltip
-                    title={isConnected ? '' : t('error.scanner-not-connected')}
-                  >
-                    <Box>
-                      <IconButton
-                        disabled={isScanning || !isConnected}
-                        onClick={scanBarcode}
-                        icon={
-                          isScanning ? (
-                            <CircularProgress size={20} color="secondary" />
-                          ) : (
-                            <ScanIcon />
-                          )
-                        }
-                        label={t('button.scan')}
-                      />
-                    </Box>
-                  </Tooltip>
-                )}
-              </Box>
+              <NumericTextInput
+                autoFocus
+                disabled={!packEditable}
+                width={160}
+                value={draft.volumePerPack ?? 0}
+                onChange={volumePerPack => onUpdate({ volumePerPack })}
```

Comment:
Can't enter decimal, and only shows 2dp..:
<img width="769" height="750" alt="Screenshot 2025-08-01 at 11 00 25 AM" src="https://github.com/user-attachments/assets/efc21be1-7768-4735-a711-2e06b3279383" />

<img width="397" height="116" alt="Screenshot 2025-08-01 at 10 57 03 AM" src="https://github.com/user-attachments/assets/767af746-a712-4eaa-ad41-839979e54258" />